crates/storage/src/storage.rs

```diff
@@ -36,50 +38,149 @@ struct Inner {
     tx_dispatcher: watch::Sender<Snapshot>,
     tx_state: Arc<watch::Sender<Snapshot>>,
     snapshots: RwLock<SnapshotCache>,
+    multistore_config: MultistoreConfig,
     #[allow(dead_code)]
     /// A handle to the dispatcher task.
+    /// This is used by `Storage::release` to wait for the task to terminate.
```

Comment:
if it's used, why do we need `allow(dead_code)`?

Reply:
It's gated on `#[cfg(test)]`. This is used to implement the async drop that we use to await shutting down the dispatcher task and releasing the RocksDB lock.